executor/stubs.go

```diff
@@ -43,7 +45,26 @@ func MountStubsCleaner(dir string, mounts []Mount) func() {
 			if st.Size() != 0 {
 				continue
 			}
-			os.Remove(p)
+			// Back up the timestamps of the dir for reproducible builds
+			// https://github.com/moby/buildkit/issues/3148
+			dir := filepath.Dir(p)
+			dirSt, err := os.Stat(dir)
+			if err != nil {
+				logrus.WithError(err).Warnf("Failed to stat %q (parent of %q)", dir, p)
+				continue
+			}
+			atime := atime.Get(dirSt)
```

Comment:
Wondering if we need the new dependency on https://github.com/djherbis/atime. Don't we have something for this already? For example, I see we have https://github.com/containerd/continuity/blob/b4ca35286886296377de39e6eafd1affae019fc3/fs/stat_atim.go#L27-L30

In containerd/continuity (but perhaps we have something elsewhere already)

Reply:
Thanks 👀 , fixed